en/guide/taking_off_landing.md

```diff
@@ -0,0 +1,220 @@
+    std::cout << "Vehicle not ready to arm ..." << std::endl;
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+}
+```
+
+The code fragment below performs the same task, but additionally exits the app if calibration is required reports what conditions are still required before the vehicle is "healthy":
+
+```cpp
+// Exit if calibration is required
+Telemetry::Health check_health = device.telemetry().health();
+bool calibration_required = false;
+if (!check_health.gyrometer_calibration_ok) {
+    std::cout << ERROR_CONSOLE_TEXT << "Gyro requires calibration." << NORMAL_CONSOLE_TEXT << std::endl;
+    calibration_required=true;
+}
+if (!check_health.accelerometer_calibration_ok) {
+    std::cout << ERROR_CONSOLE_TEXT << "Accelerometer requires calibration." << NORMAL_CONSOLE_TEXT << std::endl;
+    calibration_required=true;
+}
+if (!check_health.magnetometer_calibration_ok) {
+    std::cout << ERROR_CONSOLE_TEXT << "Magnetometer (compass) requires calibration." << NORMAL_CONSOLE_TEXT << std::endl;
+    calibration_required=true;
+}
+if (!check_health.level_calibration_ok) {
+    std::cout << ERROR_CONSOLE_TEXT << "Level calibration required." << NORMAL_CONSOLE_TEXT << std::endl;
+    calibration_required=true;
+}
+if (calibration_required) {
+    return 1;
+}
+
+
+// Check if ready to arm (reporting status)
+while (device.telemetry().health_all_ok() != true) {
+    std::cout << ERROR_CONSOLE_TEXT << "Vehicle not ready to arm. Waiting on:" << NORMAL_CONSOLE_TEXT << std::endl;
+    Telemetry::Health current_health = device.telemetry().health();
+    if (!current_health.global_position_ok) {
+        std::cout << ERROR_CONSOLE_TEXT << "  - GPS fix." << NORMAL_CONSOLE_TEXT << std::endl;
+    }
+    if (!current_health.local_position_ok) {
+        std::cout << ERROR_CONSOLE_TEXT << "  - Local position estimate." << NORMAL_CONSOLE_TEXT << std::endl;
+    }
+    if (!current_health.home_position_ok) {
+        std::cout << ERROR_CONSOLE_TEXT << "  - Home position to be set." << NORMAL_CONSOLE_TEXT << std::endl;
+    }
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+}
+```
+
+> **Note** Vehicle health can also be checked using [Telemetry:health_all_ok_async()](../api_reference/classdronecore_1_1_telemetry.md#classdronecore_1_1_telemetry_1a83b384cd04b2ed17db805cfad8bafab5). 
+
+
+### Arming
+
+Once the vehicle is ready, use the following synchronous code to arm:
+```cpp
+// Arm vehicle
+std::cout << "Arming..." << std::endl;
+const Action::Result arm_result = device.action().arm();
+
+if (arm_result != Action::Result::SUCCESS) {
+    std::cout << "Arming failed:" 
+      << Action::result_str(arm_result) 
+      <<  std::endl;
+    return 1; //Exit if arming fails
+}
+```
+
+Use [Telemetry::armed()](../api_reference/classdronecore_1_1_telemetry.md#classdronecore_1_1_telemetry_1a0ca7da7922c22509ce6d55d4ad19bcf7) to check that the vehicle has armed before calling the takeoff command.

```

Comment:
Originally I just had a timer here. Then I tried using FlightMode::Ready - but during startup my SITL showed FlightMode::Hold, so decided just checking `armed` is more robust. 

@julianoes - use this in example code?

Reply:
@julianoes OK, I will remove arm code.

Should we update example code to include the calibration checks? I'd argue yes, because it is best practice.  What about also adding the more verbose status info - should that also be added to example code.

Reply:
Yes, that would be nice.